src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -2072,30 +2072,30 @@ private int deactivate(WorkQueue w, int r, int phase) {
                 }
             }
             if (p != phase &&                    // check quiescent termination
-                ((qc & RC_MASK) > 0L || quiescent() <= 0)) {
+                ((runState & SHUTDOWN) == 0L || quiescent() <= 0)) {
```

Comment:
@DougLea `runState` is a volatile, so this change implies a slight difference in memory effects from the `qc` check, I presume this is intentional? 🤔

Reply:
Yes. This check makes termination a bit faster because SHUTDOWN is also true if STOP, (as quiescent() will report)  so avoids enough spins and rechecks to be worth the extra read.